File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int curr = nums[0]; 
        int start = 1; 
        for(int i = start; i < nums.length; i++){
            if(nums[i] != curr){
                nums[start++] = nums[i];
                curr = nums[i]; 
            }
        }
        return start; 
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        set<int> hashSet; 
        int index = 0; 
        for(int i = 0; i < nums.size(); i++){
            if(!hashSet.count(nums[i])){
                nums[index++] = nums[i]; 
            }
            hashSet.insert(nums[i]); 
        }
        return index; 
    }
};